### PR TITLE
[no-relnotes] Fix release scripts

### PR DESCRIPTION
**Pull request by elezar**
These changes address issues with the release process.

### CODE REVIEW

File: hack/generate-changelog.sh [hunk 1]
```diff
@@ -64,11 +57,16 @@ while [[ $# -gt 0 ]]; do
 done
 
 # Fetch the latest tags from the remote
-git fetch $REMOTE --tags
+git fetch --tags
```
Comment (ArangoGutierrez) [inline hack/generate-changelog.sh:60]: this means, this script is never to be run manually

**Review by ArangoGutierrez — COMMENTED** [review]
one question

File: hack/generate-changelog.sh (hunk 1 above)
Comment (elezar) [reply hack/generate-changelog.sh:60]: Can you confirm what `git fetch --tags` does on your environment.

Comment (ArangoGutierrez) [reply hack/generate-changelog.sh:60]: I see, I am wrong.

**Review by ArangoGutierrez — APPROVED** [review]

File: hack/generate-changelog.sh [hunk 2]
```diff
@@ -64,11 +57,17 @@ while [[ $# -gt 0 ]]; do
 done
 
 # Fetch the latest tags from the remote
-git fetch $REMOTE --tags
+remote=$( git remote -v | grep -E "NVIDIA/k8s-device-plugin(\.git)?\s" | grep -oE "^[a-z]+" | sort -u )
```
Comment (ArangoGutierrez) [inline hack/generate-changelog.sh:60]:
```bash
❯ git remote -v | grep -E "NVIDIA/k8s-device-plugin(\.git)?\s" | grep -oE "^[a-z]+" | sort -u
upstream
```

**Review by ArangoGutierrez — APPROVED** [review]

File: hack/update-helm-index.sh [hunk 3]
```diff
@@ -88,8 +88,9 @@ else
 			echo "Skipping $asset_url, does not look like a Helm chart archive"
 			continue
 		fi
-		echo "Downloading $asset_url..."
-		curl -sSfLO -o $HELM_REPO_PATH/stable/$(basename $asset_url) $asset_url
+		target=$HELM_REPO_PATH/stable/$(basename $asset_url)
```
Comment (ArangoGutierrez) [inline hack/update-helm-index.sh:91]: This fixes the `update-helm-index`, tested locally.
Thanks for this fix @elezar

File: hack/update-helm-index.sh [hunk 4]
```diff
@@ -82,21 +82,7 @@ if [ -n "$asset_local" ]; then
...
-		echo "Downloading $asset_url..."
-		curl -sSfLO -o $HELM_REPO_PATH/stable/$(basename $asset_url) $asset_url
-		# We rely on all release assets having the same baseurl
-		download_baseurl=`dirname $HELM_REPO_PATH/stable/$(basename $asset_url)`
-	done
-	if [ -z "$download_baseurl" ]; then
-		echo "No Helm chart release assets found"
-		exit 0
-	fi
+	curl -sSfO --output-dir "${HELM_REPO_PATH}/stable" ${asset_urls}
```
Comment (ArangoGutierrez) [inline hack/update-helm-index.sh:85]: tested locally, it works

File: hack/generate-changelog.sh (hunk 2 above)
Comment (elezar) [reply hack/generate-changelog.sh:60]: Yes, this is intended to get the remote to use for fetching the tags. This should address us not having `origin` as our default remote.

File: hack/prepare-release.sh [hunk 5]
```diff
@@ -137,10 +137,40 @@ else
...
+fi
+
+# Create a release issue.
+echo "Creating release tracking issue"
+cat RELEASE.md | sed "s/{{ .VERSION }}/$release/g" | \
+    gh issue create -F - \
+        -R NVIDIA/cloud-native-team \
+        --title "Release k8s-device-plugin $release" \
+        --label release
+
```
Comment (ArangoGutierrez) [inline hack/prepare-release.sh:162]: trailing new line

File: hack/prepare-release.sh [hunk 6]
```diff
@@ -154,11 +184,20 @@ echo Patching deployments/helm/Chart.yaml to refer to $semver
...
+    git commit -s -m "Bump version to $release in README"
+else
+    echo "Skipping README update for prerelease version"
 fi
-# Patch README.md
-echo Patching README.md to refer to $release
-$SED -E -i -e "s/([^[:space:]])$previous_version([^[:alnum:]]|$)/\1$release\2/g" README.md
-$SED -E -i -e "s/$pre_semver/$semver/g" README.md
+
+echo "Please validated changes and create a pull request"
```
Comment (ArangoGutierrez) [inline hack/prepare-release.sh:203]: missing new line

File: RELEASE.md [hunk 7]
```diff
@@ -4,23 +4,43 @@ The device plugin consists in two artifacts:
...
-# Release Process Checklist
-- [ ] Run the `./hack/prepare-release.sh` script to update the version in all the needed files
-- [ ] Run the `./hack/package-helm-charts.sh` script to generate the helm charts
-- [ ] Run the `./hack/generate-changelog.sh` script to generate the a draft changelog
-- [ ] Update the CHANGELOG.md file with the generated changelog
-- [ ] Commit, Tag and Push to Gitlab
-- [ ] Wait for the `Release` GitHub Action to complete
+# Release Process Checklist:
+- [ ] Create a release PR:
+    - [ ] Run the `./hack/prepare-release.sh` script to update the version in all the needed files.
```
Comment (ArangoGutierrez) [inline RELEASE.md:9]: running `prepare-release` will create an issue, we should mention that here

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: hack/prepare-release.sh (hunk 6 above)
Comment (elezar) [reply hack/prepare-release.sh]: Sorry. Had a local change. Pushed now.

**Review by ArangoGutierrez — APPROVED** [review]

**Pull request merged**